src/Reader/STLReader.php

```diff
@@ -22,12 +22,22 @@ public static function forFile($fileName)
 
     private static function getSTLType($handle)
     {
-        $type = self::BINARY;
-        if (strtolower(fread($handle, 5)) == 'solid') {
-            $type = self::TEXT;
+        $type = self::TEXT;
+        if (self::binaryHasExpectedLength($handle)) {
```

Comment:
> There is also a very unlikely edge case when the bytes 80-84 of an ASCII STL file, produce an expected filesize that is equal to the actual filesize, causing the file to be wrongly interpreted as binary, but that is extremely unlikely.

Another edge case is when a binary STL with the incorrect triangle count will be detected as ASCII. Maybe add one more check here?

`if (strtolower(fread($handle, 5)) != 'solid' || self::binaryHasExpectedLength($handle)) {`